### PR TITLE
feature: add data filtering framework

### PR DESCRIPTION
**Pull request by rambohe-ch**
/kind feature

#### What this PR does / why we need it:
add data filtering framework
1. support endpointslice filter for keeping service traffic in-bound of nodePool
2. support master service mutation for pod use InClusterConfig access kube-apiserver

you can check the proposal here: https://github.com/openyurtio/openyurt/pull/388

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/assign @Fei-Guo 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/yurthub/filter/initializer/initializer.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	SetStorageWrapper(s cachemanager.StorageWrapper) error
+}
+
+// WantsMasterServiceAddr is an interface for setting mutated master service address
+type WantsMasterServiceAddr interface {
+	SetMasterServiceAddr(addr string) error
+}
+
+// filterInitializer is responsible for initializing filter
+type filterInitializer struct {
```
Comment (Fei-Guo) [inline pkg/yurthub/filter/initializer/initializer.go:58]: Suggest to rename this to "genericFilterInitializer".

**Review by Fei-Guo — COMMENTED** [review]
Overall, LGTM.

We need to add tests in the future change.

File: pkg/yurthub/filter/initializer/initializer.go (hunk 1 above)
Comment (rambohe-ch) [reply pkg/yurthub/filter/initializer/initializer.go]: fixed

**Pull request merged**